### DISCUSSION TITLE
argo wf api Qs

### DISCUSSION THREAD

**Discussion by tooptoop4** [argoproj/argo-workflows#7712] [Q&A]
"argo-linux-amd64 auth token" generates token that works in remote curl request against /api/v1/workflows/auth .etc
some questions:
1. how long is the token valid for?
2. can the "argo-linux-amd64 auth token" command be run only on host with kubectl access to argo server? or can any arbitrary host generate a validate token?
3. since no credentials are passed to "argo-linux-amd64 auth token" command, how to differentiate between having read vs write access on the api?
4. can the api integrate with LDAP/SSO credentials or similar?

**Comment by sarabala1979** [comment]
how long is the token valid for?
if you don't have  any exec on your kubeconfig, Kubeconfig user/serviceaccount token will be return. it is valid for long

can the "argo-linux-amd64 auth token" command be run only on host with kubectl access to argo server? or can any arbitrary host generate a validate token?
argo-cli will communicate through your kubeconfig. 
since no credentials are passed to "argo-linux-amd64 auth token" command, how to differentiate between having read vs write access on the api?
tokens are based your kubeconfig.  API will work based  on  your RBAC 
can the api integrate with LDAP/SSO credentials or similar?
no
